### ISSUE TITLE
We need a way to do a layout where columns auto-fit

### ISSUE THREAD

**Issue by Samitier** [holaluz/margarita#405]
## What's the problem?

We have a missing feature in MaLayout that we desperately need. It's basically the "`display: flex` experience" 😂 

We need a way to set up columns that automatically fit their own width inside a row, like:

![image](https://user-images.githubusercontent.com/4160121/116520647-48549280-a8d3-11eb-9412-52dcb90dd9a4.png)

Right now, the closest thing we have in MaLayout is grid-auto-flow (the `*` operator), which always fills the whole width of the row, no matter the content of the columns.

![image](https://user-images.githubusercontent.com/4160121/116522377-6cb16e80-a8d5-11eb-87fc-d10561623498.png)

I would like to not change the API of MaLayout or having to create a different component for this, and I think we can get something working if we play with the `justify` prop.

## What do we do to fix it?

What if the `grid-auto-flow` operator works differently, depending on the `justify` prop? 

So basically this:

![image](https://user-images.githubusercontent.com/4160121/116523045-17299180-a8d6-11eb-8d72-3ffa10890648.png)

What do you think? It will be a breaking change for sure 😞

**Comment by Samitier** [comment]
hmm if we use `justify: center` we will lose the opportunity to center the row, so maybe it's not a good idea. Maybe we need a new value for justify

**Comment by edufarre** [comment]
What about allowing the `columns` property to have a value like `max-content`, thus keeping the same way approach as using `css grid` and not mixing different properties behaviour.

**Comment by afontcu** [comment]
What about `<ma-layout>` without `columns` prop? Too magical? 😂

**Comment by sergigraciaj** [comment]
> hmm if we use justify: center we will lose the opportunity to center the row, so maybe it's not a good idea. Maybe we need a new value for justify

Totally, `justify: expand`, ` justify:fit`? Or just boolean fit prop?

**Comment by Samitier** [comment]
> What about allowing the `columns` property to have a value like `max-content`, thus keeping the same way approach as using `css grid` and not mixing different properties behaviour.

I like how it sounds, but it adds even more complexity for an already complex prop `columns` IMHO. I didn't want to touch that because I think it's on the verge to become unreadable 😂 .


> What about `<ma-layout>` without `columns` prop? Too magical? 😂 It'd be similar to Stack (it has no "columns/row" props, but it lays content vertically).

Then we wont be able to build it with other rows with a single `ma-layout`. For example, if we want a 12 col row before that one, we will need to do:
```html
<ma-stack>
   <div />
   <ma-layout />
</ma-stack>
```
instead of:
```html
<ma-layout columns="12 - *">
    ...
</ma-layout>
```

> Totally, justify: expand, justify:fit? Or just boolean fit prop?

I like this! But then, what happens for other cases, for example if we have fixed columns that doesn't add up to 12:

```html
<ma-layout columns="3" justify="fit">
</ma-layout>
```
What does this "3" column do? Ignores the justify and stays in the left occupying "3"? Or ignores its own width and "fits" to the content of its children? Or it fits itself to the content of its children with a max width of "3" (I kinda like this one tbh 😂 ).

**Comment by sergigraciaj** [comment]
I would say `justify: fit` to ignore columns, if it fits content with a max width of the columns wouldn't that lead us to unexpected behaviours? 🤔

**Comment by afontcu** [comment]
any update on this? 🙌

**Comment by Samitier** [comment]
Hello, as we said in the frontend crew, we will:

 - Fix the `*` operator so that it sets n columns of the same size. This will be a breaking change, but as we said we think it should have been the expected behaviour of this component since the beginning.
 
 - Create a `ma-column` component (or `ma-inline`, or `ma-flex`, whatever) that lets us position content in the X axis, without taking into account the 12 col layout. Exactly like a `display: flex` or a `display: inline-block` would do. We should do a RFC to make sure we take into account all of the use cases.
